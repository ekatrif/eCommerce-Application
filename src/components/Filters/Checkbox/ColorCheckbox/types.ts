export interface IColor {
  color: {
    id: number;
    checked: boolean;
    label: string;
  };
  handleChangeCheckedColor: (id: number) => void;
}
