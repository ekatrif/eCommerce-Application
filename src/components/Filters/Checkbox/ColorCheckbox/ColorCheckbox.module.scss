.root:checked {
      color: '#000';
}

.wrap {
    width: 100%;
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
    align-items: center;
    margin-left: 0;
};

.label {
    font-size: .8rem,
};