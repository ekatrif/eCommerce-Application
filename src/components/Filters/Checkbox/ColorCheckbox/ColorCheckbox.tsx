import React from 'react';
import FormGroup from '@mui/material/FormGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import { IColor } from './types';

const ColorCheckbox: React.FC<IColor> = ({ handleChangeCheckedColor, color }) => {
  const { checked, label, id } = color;
  return (
    <FormGroup>
      <FormControlLabel
        control={<Checkbox checked={checked} onChange={(): void => handleChangeCheckedColor(id)} />}
        label={label}
      />
    </FormGroup>
  );
};

export default ColorCheckbox;
