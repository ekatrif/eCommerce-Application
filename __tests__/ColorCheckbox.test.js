import React from 'react';
import renderer from 'react-test-renderer';
import ColorCheckbox from '../src/components/Filters/Checkbox/ColorCheckbox/ColorCheckbox'

test('ColorCheckbox renders correctly', () => {
  const tree = renderer
    .create(<ColorCheckbox key="0" color="testColor" handleChangeCheckedColor="" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
