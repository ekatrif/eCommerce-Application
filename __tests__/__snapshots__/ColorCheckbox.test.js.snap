// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCheckbox renders correctly 1`] = `
<div
  className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
>
  <label
    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
  >
    <span
      className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <input
        className="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </label>
</div>
`;
